tests/map_macro.rs

```diff
@@ -1,19 +1,30 @@
 #![recursion_limit="128"]
 
-extern crate futures;
+extern crate futures_core;
+extern crate futures_executor;
+extern crate futures_util;
 
 #[macro_use]
 extern crate futures_signals;
 
-
 #[macro_export]
 macro_rules! map_tests {
     ($name:ident, ($($ref:tt)+), ($($arg:tt)+)) => {
         #[cfg(test)]
         mod $name {
-            use futures::Async;
+            use futures_core::Poll;
+            use futures_executor::block_on;
+            use futures_util::future::poll_fn;
+
             use futures_signals::signal::{Signal, always};
 
+            // NB it's not a problem to block on an Always because they always return Async::Ready!
+            fn poll_always<S: Signal>(signal: &mut S) -> Option<S::Item> {
```

Comment:
Why doesn't this use the same `with_noop_context` system as above?

Reply:
Happy to do so; I actually wrote `poll_always` first and didn't feel I necessarily needed to change it.

If I were to re-use `with_noop_context` I could either copy and paste or export it so this test can access it (perhaps in `mod internal`). Do you have a preference on either of these solutions?

Reply:
I guess an option c) is also to move the other test into the `tests` directory, but I prefer inline unit tests to integration tests wherever possible. Up to you 😄

Reply:
I actually prefer integration tests. They're more work to setup, but they ensure that the tests work *from the end user's perspective*. It's easy to accidentally rely upon internal private stuff with inline tests.

You don't need to worry about this right now, though, since I'll be making some changes to the tests anyways, so I'll just accept this PR. Thanks again!